starboard/shared/win32/posix_emu/dirent.cc

```diff
@@ -180,18 +190,26 @@ DIR* opendir(const char* path) {
 }
 
 int closedir(DIR* dir) {
+  if (!dir) {
+    errno = EBADF;
+    return -1;
+  }
   bool success = CloseHandle(dir->handle);
   handle_db_get(dir->fd, true);
   delete dir;
-  return success ? 0 : -1;
+  if (!success) {
+    errno = EINTR;

```

Comment:
This one looks a little random - is it supposed to be EINTR or something more fatal ?

Reply:
https://pubs.opengroup.org/onlinepubs/7908799/xsh/errors.html 
I couldn't figure out what else could work instead, there's not really a catch all fatal error that I could find